### PR TITLE
Feature/develop mode

### PR DESCRIPTION
**Pull request by memsharded**
https://github.com/conan-io/conan/issues/1952

### CODE REVIEW

File: conans/client/manager.py [hunk 1]
```diff
@@ -110,21 +110,41 @@ def __init__(self, client_cache, user_io, runner, remote_manager, search_manager
...
 
         return conanfile
 
+    def _load_install_conanfile(self, loader, reference_or_path, conanfile_filename, cwd=None):
+        """loads a conanfile for installation: install, info
+        cwd only used for virtuals, to pass it the current directory and make available the
+        conanfile.conanfile_directory (smell)"""
+        if isinstance(reference_or_path, ConanFileReference):
+            conanfile = loader.load_virtual([reference_or_path], cwd)
+            loader.dev_reference = reference_or_path
```
Comment (lasote) [inline conans/client/manager.py]: So `conan install ref` should consider `ref` as a package being developed?

File: conans/test/functional/develop_test.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        self.assertIn("Develop package_id!", client.out)
+        consumer = """from conans import ConanFile
+class Pkg(ConanFile):
+    requires = "Pkg/0.1@user/testing"
+"""
+        client.save({"conanfile.py": consumer})
+        client.run("create Other/1.0@user/testing")
+        self.assertNotIn("Develop", client.out)
+
+    def local_commands_test(self):
```
Comment (lasote) [inline conans/test/functional/develop_test.py]: Add a test to `conan install ref`, what should happen?

File: conans/client/manager.py (hunk 1 above)
Comment (memsharded) [reply conans/client/manager.py]: No, fixed

File: conans/test/functional/develop_test.py (hunk 2 above)
Comment (memsharded) [reply conans/test/functional/develop_test.py]: Test added

**Pull request merged**